pkg/cmd/secret/shared/shared.go

```diff
@@ -1,9 +1,91 @@
 package shared
 
+import (
+	"errors"
+	"strings"
+)
+
 type Visibility string
 
 const (
 	All      = "all"
 	Private  = "private"
 	Selected = "selected"
 )
+
+type App string
+
+const (
+	Actions    = "actions"
+	Codespaces = "codespaces"
+	Dependabot = "dependabot"
+	Unknown    = "unknown"
+)
+
+func (app App) String() string {
+	return string(app)
+}
+
+func (app App) Title() string {
+	return strings.Title(app.String())
+}
+
+type SecretEntity string
+
+const (
+	Repository   = "repository"
+	Organization = "organization"
+	User         = "user"
+	Environment  = "environment"
+)
+
+func GetSecretEntity(orgName, envName string, userSecrets bool) (SecretEntity, error) {
+	orgSet := orgName != ""
+	envSet := envName != ""
+
+	if orgSet && envSet || orgSet && userSecrets || envSet && userSecrets {
+		return "", errors.New("cannot specify multiple secret entities")
+	}
+
+	if orgSet {
+		return Organization, nil
+	}
+	if envSet {
+		return Environment, nil
+	}
+	if userSecrets {
+		return User, nil
+	}
+	return Repository, nil
+}
+
+func GetSecretApp(app string, entity SecretEntity) App {
```

Comment:
Is there a good reason to support an `Unknown` app type here? Or do you think it makes more sense to just return an error?

Reply:
An error makes sense here 👍. I kept `Unknown` to represent a "nil" App

https://github.com/cli/cli/pull/5134/commits/4b0c4de225fc89283dc9fca8c65abed9ed2a7c40